community/codegen/src/main/java/org/neo4j/codegen/ExpressionVisitor.java

```diff
@@ -41,15 +41,37 @@
 
     void ternary( Expression test, Expression onTrue, Expression onFalse );
 
-    void eq( Expression lhs, Expression rhs );
+    void ternaryOnNull( Expression test, Expression onTrue, Expression onFalse );
+
+    void ternaryOnNonNull( Expression test, Expression onTrue, Expression onFalse );
+
+    void equal( Expression lhs, Expression rhs, TypeReference type );
 
     void or( Expression lhs, Expression rhs );
 
-    void add( Expression lhs, Expression rhs );
+    void addInts( Expression lhs, Expression rhs );
+
+    void addLongs( Expression lhs, Expression rhs );
+
+    void addDoubles( Expression lhs, Expression rhs );
+
+    void gt( Expression lhs, Expression rhs, TypeReference type );
+
+    void subtractInts( Expression lhs, Expression rhs );
 
-    void gt( Expression lhs, Expression rhs );
+    void subtractLongs( Expression lhs, Expression rhs );
 
-    void sub( Expression lhs, Expression rhs );
+    void subtractDoubles( Expression lhs, Expression rhs );
+
+    void multiplyLongs( Expression lhs, Expression rhs );
+
+    void multiplyDoubles( Expression lhs, Expression rhs );
 
     void cast( TypeReference type, Expression expression );
+
+    void newArray( TypeReference type, Expression... constants );
+
+    void longToDouble( Expression expression );
+
+    void pop( Expression expression );
```

Comment:
The reason I went looking was because I didn't like the name...

The purpose of this method would be to evaluate the expression for it's side effects only, and we already have methods for that, on the _Statement_ level.